replication/event.go

```diff
@@ -229,15 +229,59 @@ type PreviousGTIDsEvent struct {
 	GTIDSets string
 }
 
+type GtidFormat int
+
+const (
+	GtidFormatClassic = iota
+	GtidFormatTagged
+)
+
+func decodeSid(data []byte) (format GtidFormat, sidnr uint64) {
+	if data[7] == 1 {
+		format = GtidFormatTagged
+	}
+
+	if format == GtidFormatTagged {
+		sid_mask := []byte{0, 255, 255, 255, 255, 255, 255, 0}
+
+		// Apply the mask
+		for i, _ := range data[:8] {
+			data[i] &= sid_mask[i]
+		}
+		data = append(data, 0)
+
+		// sidnr
+		sidnr = binary.LittleEndian.Uint64(data[1:])
+		return
+	}
+	sidnr = binary.LittleEndian.Uint64(data)
+	return
+}
+
 func (e *PreviousGTIDsEvent) Decode(data []byte) error {
 	pos := 0
-	uuidCount := binary.LittleEndian.Uint16(data[pos : pos+8])
+
+	gtidinfo := make([]byte, 8)
+	copy(gtidinfo, data[:8])
```

Comment:
I see that the new copy is caused by `decodeSid` will modify data in-place. How about let `decodeSid` copy the needed data? So there's no risk that in future we need to change the code and caller forget to copy it.

Reply:
yes good one.